staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -37,29 +35,29 @@
 
     public BatchPathProvider(final String basePath)
     {
-        this.basePath = Paths.get(basePath);
+        this.basePath = Paths.get(basePath).normalize();
     }
 
-    public Path getPathForTenant(final TenantId tenantId) throws InvalidTenantIdException
+    public Path getPathForTenant(final TenantId tenantId)
```

Comment:
You shouldn't be introducing checked exceptions for impossible situations.  A checked exception is telling the caller that they must deal with this exception, but the the tenant id should already have been validated during the constructor, so the caller should not have to deal with this.  This is the correct situation for using an unchecked exception.